.github/workflows/update-deps.yaml

```diff
@@ -0,0 +1,50 @@
+name: Auto-update Python Deps
+on:
+  # Manual trigger
+  workflow_dispatch:
+  # Check for new deps every Tuesday at 12:00
+  schedule:
+    - cron: "0 12 * * 2"
+
+jobs:
+  update-lib:
+    name: Check deps
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Install dependencies
+        run: |
+          curl -LsSf https://astral.sh/uv/install.sh | sh
+          source $HOME/.cargo/env
+          uv venv && source .venv/bin/activate
+          uv pip install -r requirements-dev.txt
+
+      - name: Run update
+        run: |
+          uv run tox -e update-dependencies
+          uv run tox -e generate-requirements
```

Comment:
I'm using this idiom in my local:

`uvx -p 3.10 tox -e lint`

It works great, installs Python on demand, etc.

There's also `uvx --with-requirements req.txt ...` flag which can come in handy.

Reply:
Thanks! :)